### PR TITLE
ADR Generator

### PR DESCRIPTION
**Pull request by samsonasu**

### CODE REVIEW

File: lib/jefferies_tube/railtie.rb [hunk 1]
```diff
@@ -4,10 +4,13 @@
 require 'rails'
 
 module JefferiesTube
-
   class Railtie < ::Rails::Railtie
     railtie_name :jefferies_tube
 
+    # generators do
```
Comment (tdipiazza) [inline lib/jefferies_tube/railtie.rb:10]: Why is this commented out? Is it not needed?

Comment (samsonasu) [reply lib/jefferies_tube/railtie.rb:10]: no, good catch.  it works through magic once i got the folders / class names correct but was a classically difficult to figure out rails magic thing

File: lib/generators/rails/adr_generator.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      def create_adr
+        @adr_dir = Rails.root.join('doc', 'adr')
+        @padded_number = sprintf("%03d", get_new_number)
+        slug = name.downcase.underscore.gsub(/[^a-zA-Z0-9]/, "-")
+        @title = name.titlecase
+        template "adr.md.erb", @adr_dir.join("#{@padded_number}-#{slug}.md"), binding
+      end
+
+      private
+      def get_new_number
```
Comment (tdipiazza) [inline lib/generators/rails/adr_generator.rb:18]: There's a possibility we'll end up with duplicate numbers if two people decide to create a new ADR on the same project at the same time, but I guess the inconvenience of having to resolve that manually is outweighed by the convenience of auto-numbering

**Comment by tdipiazza** [comment]
🚢

**Pull request merged**